src/lib-components/lists/DynamicTable.vue

```diff
@@ -95,22 +126,16 @@ const { columns, hasActions, isEmptyCellValue, rows } = useTable(
   publicMethods
 )
 
-const currentSort = ref(
-  props.tableOptions.defaultSort ? props.tableOptions.defaultSort : ""
-)
-const currentSortDirection = ref(
-  props.tableOptions.defaultSortDirection
-    ? props.tableOptions.defaultSortDirection
-    : "desc"
-)
+const currentSort = ref(props.config.sort?.defaultSort)
+const currentSortDirection = ref(props.config.sort?.defaultSortDir)
 const dateRange = ref({
   minDate: 0,
   maxDate: 0,
 })
 
 const pagination = ref({
   page: 1,
-  perPage: 10,
+  perPage: props.config.pagination?.perPage || 10,
```

Comment:
@steven-klein example where the compiler yelled about me about `config.pagination` being potentially `undefined`, which I didn't expect. I end up having to set the default again, which maybe means we just don't use `withDefaults`?